### PR TITLE
Add functorch-based regression training example.

### PR DESCRIPTION
**Pull request by dellis23**

### CODE REVIEW

File: examples/regression.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    #
+    # Training
+    #
+    print("Compiling training function with Torch-MLIR")
+    train_args = (w, b, X_test, y_test)
+    graph = functorch.make_fx(train)(*train_args)
+
+    # BEFORE SUBMIT: Explain why we need to do this?  Embed this into
+    # torch_mlir.compile as an argument?  Make it the default?
+    strip_overloads(graph)
```
Comment (ramiro050) [inline examples/regression.py:82]: I think this is probably fine to include in `torch_mlir.compile`. You just have to make sure that the input graph is an instance of `torch.fx.GraphModule` before passing it to `strip_overloads`. This should be safe to call on any FX graph

File: examples/regression.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+def loss_fn(w, b, X, y):
+    y_pred = forward(w, b, X)
+    return mse(y_pred, y)
+
+
+grad_fn = functorch.grad(loss_fn, argnums=(0, 1))
+
+
+def update(w, b, grad_w, grad_b):
+    new_w = w - grad_w * 0.05
```
Comment (silvasean) [inline examples/regression.py:58]: Nit: put 0.05 into a variable "learning rate" for clarity

File: examples/regression.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    graph = functorch.make_fx(train)(*train_args)
+
+    # BEFORE SUBMIT: Explain why we need to do this?  Embed this into
+    # torch_mlir.compile as an argument?  Make it the default?
+    strip_overloads(graph)
+
+    linalg_on_tensors_mlir = torch_mlir.compile(
+        graph,
+        train_args,
+        output_type=torch_mlir.OutputType.LINALG_ON_TENSORS,
```
Comment (silvasean) [inline examples/regression.py:87]: This can be "linalg-on-tensors" for simplicity.

File: examples/regression.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        use_tracing=False)
```
Comment (silvasean) [inline examples/regression.py:88]: nit: remove -- use_tracing=False is the default

File: examples/regression.py (hunk 1 above)
Comment (silvasean) [reply examples/regression.py:82]: Yes, let's do that.

File: examples/regression.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    print("Training on IREE")
+    for _ in range(30):
+        w, b = invoker.forward(*train_args)
+        train_args = (w, b, X_test, y_test)
+        print("Loss:", loss_fn(w, b, X_test, y_test))
+    print()
+
+    #
+    # Inference
+    #
```
Comment (silvasean) [inline examples/regression.py:104]: maybe set a variable `inference_args = train_args[:3] # Remove the labels for inference.`

**Review by silvasean — APPROVED** [review]
Awesome!!

File: examples/regression.py (hunk 2 above)
Comment (dellis23) [reply examples/regression.py]: Done.

File: examples/regression.py (hunk 3 above)
Comment (dellis23) [reply examples/regression.py:88]: Thanks, much nicer than having to look up the enum every time.

File: examples/regression.py (hunk 4 above)
Comment (dellis23) [reply examples/regression.py:89]: Done.
